README.md

```diff
@@ -65,3 +65,73 @@ podman-compose -f deploy/podman/observability.yaml up
 ```
 
 The Grafana and Prometheus web UIs are then accessible on `http://localhost:3000` and `http://localhost:9090`, respectively.
+
+
+## Agent Image Creation
+
+This section aims to provide a comprehensive guide to build a bootable container image for the agent. This means using a bootable Fedora container image as base, we will create our own image that will allow users to enroll devices (VMs or physical appliances) into their own FlightCtl service.
+
+Let's take a look at the provided Containerfile located in packaging/Containerfile.fedora:
+
+```
+FROM quay.io/centos-bootc/fedora-bootc:eln
+
+COPY rpmbuild/RPMS/x86_64/flightctl-agent-0.0.1-1.el9.x86_64.rpm /tmp/
+
+COPY packaging/flightctl-custom-assets/flightctl_rsa.pub /usr/etc-system/root.keys
+RUN touch /etc/ssh/sshd_config.d/30-auth-system.conf; \
+    mkdir -p /usr/etc-system/; \
+    echo 'AuthorizedKeysFile /usr/etc-system/%u.keys' >> /etc/ssh/sshd_config.d/30-auth-system.conf; \
+    chmod 0600 /usr/etc-system/root.keys
+VOLUME /var/roothome
+
+ADD packaging/flightctl-custom-assets/config.yaml /etc/flightctl/
+ADD packaging/flightctl-custom-assets/ca.crt /etc/flightctl
+ADD packaging/flightctl-custom-assets/client-enrollment.* /etc/flightctl/
+
+RUN rpm-ostree install -y /tmp/flightctl-agent-0.0.1-1.el9.x86_64.rpm
+RUN ln -s /usr/lib/systemd/system/podman.socket /usr/lib/systemd/system/multi-user.target.wants/
+RUN ln -s /usr/lib/systemd/system/flightctl-agent.service /usr/lib/systemd/system/multi-user.target.wants/
+RUN ostree container commit 
+
+```
+
+In order to make this work for your own service, first you need to have deployed the service (by following the section above where the podman-compose manifests is applied).
+
+Execute `make rpm` to build the agent RPM package. It will be picked up and injected into the image.
+Now, add the following assets to the directory `packaging/flightctl-custom-assets/`:
+
+- The public SSH key you want to inject as `packaging/flightctl-custom-assets/flightctl_rsa.pub`
+- The CA certificate:  `packaging/flightctl-custom-assets/ca.crt`
+- The client enrollment keypair: 
+  - `packaging/flightctl-custom-assets/client-enrollment.crt`
+  - `packaging/flightctl-custom-assets/client-enrollment.key`
+- The configuration file that points to your service: `packaging/flightctl-custom-assets/config.yaml`
+
+It is important to highlight, that in a production environment, SSH will be disabled by default for security reasons. 
+The certificates will be stored probably in `$HOME/.flightctl/certs/`.
+
+The config.yaml should look like this:
+
+```
+agent:
+  server: https://localhost:3333
+  enrollementUi: https://localhost:3333
+  statusUpdateInterval: 1m0s
+  fetchSpecInterval: 1m0s
+```
+
+Change localhost by the IP of the machine where the service is exposed. Take into account that these services must be exposed to where the agent is going to be deployed.
```

Comment:
```suggestion
Change localhost by the IP/hostname and port where the service is exposed. Take into account that these services must be exposed to where the agent is going to be deployed.
```

What do you mean by "Take into account that these services must be exposed to where the agent is going to be deployed." ?

Reply:
May be... take into account these services must be accessible from where the agent is going to be deployed ?

Reply:
Hey! yes, brain switched off yesterday night... xD

I'll rephrase it, I just wanted to say that services must be reachable by the agent device.

Reply:
Using active voice may be easier to read: "Take into account that agents must be able to connect to the service."

Reply:
+1